test/Concurrency/flow_isolation.swift

```diff
@@ -729,3 +729,28 @@ actor CheckDeinitFromActor {
     ns = nil // expected-warning {{cannot access property 'ns' with a non-sendable type 'NonSendableType?' from non-isolated deinit; this is an error in Swift 6}}
   }
 }
+
+// https://github.com/apple/swift/issues/70550
+func testActorWithInitAccessorInit() {
+  @available(SwiftStdlib 5.1, *)
+  actor Angle {
+    var degrees: Double
+    var radians: Double = 0 {
+      @storageRestrictions(initializes: degrees)
+      init(initialValue)  {
+        degrees = initialValue * 180 / .pi
+      }
+
+      get { degrees * .pi / 180 }
+      set { degrees = newValue * 180 / .pi }
+    }
+
+    init(degrees: Double) {
+      self.degrees = degrees // Ok
+    }
+
+    init(radians: Double) {
+      self.radians = radians // Ok
+    }
```

Comment:
Some suggestions for other things to test for:

1. What happens if you init `radians`, escape `self` from the initializer, and then try to access `radians` again?
2.  Same as above but if `radians` is a nonisolated property.
3. What happens if you try to escape `self` prior to `radians` being initialized?

I assume that for all of these, they _should_ work the same as a computed property (which is hopefully already tested).  Only (3) is one that is really to ensure FlowIsolation doesn't crash, since it should already get flagged as an error by DI prior.

Reply:
Otherwise if those are working, then this PR LGTM.

Reply:
@kavon I had to make some changes to FlowIsolation pass to recognize `setter` use of init accessor properties. That cannot be done earlier because only DI knows when to use init accessor and when to use a setter for a particular property.